### ISSUE TITLE
HTLC SHA256 hash differs from sha256sum in Linux

### ISSUE THREAD

**Issue by MichelSantos** [bitshares/bitshares-ui#2722]
**Describe the bug**

The SHA256 of the preimage on the HTLC form differs from the sha256sum in Linux

**To Reproduce**

1. Load the wallet https://develop.bitshares.org.
2. Change the connection to one of the public testnet nodes
3. Log in with a cloud account
4. Click on Spotlight
5. Click on HTLC

The preimage that is automatically generated display a SHA256 hash of the preimage.  The values are copied and pasted below.

preimage: Dt9onxvUKMBUWmzcbS3GzLXS5iD8rs hash type: sha256

hash: 48a433cddfd442148795dad0596ef5920e6a94c9ae62cde82e85cceb5525c09c preimage size: 30

**Expected behavior**

The sha256sum hash of Dt9onxvUKMBUWmzcbS3GzLXS5iD8rs is a54f0ace96f4743d37cdc3393b565782ce2b7505a432069604f25d39ed73db67

**Desktop:**

 - OS: Linux
 - Browser Firefox
 - Version 66

**Comment by abitmore** [comment]
```
$ echo -n 'Dt9onxvUKMBUWmzcbS3GzLXS5iD8rs' | sha256sum
48a433cddfd442148795dad0596ef5920e6a94c9ae62cde82e85cceb5525c09c  -
```

**Comment by sschiessl-bcp** [comment]
@MichelSantos How did you calculate the hash exactly?

**Comment by MichelSantos** [comment]
> @MichelSantos How did you calculate the hash exactly?

@sschiessl-bcp I was testing with

1. sha256sum -t
2. Entering text, and clicking Enter
3. Ctrl+D

I also tested by placing the pre-image into a file and then calling

1. sha256sum filename

But both approaches were inserting newlines.  After ensuring that no newlines were being inserted, according to the helpful tip by @abitmore, I was able to get the same hash as in the reference wallet.

I'm sorry for wasting your time.
